src/components/map/Map.tsx

```diff
@@ -120,26 +158,61 @@ export const Map = ({
             onPress={() => {
               if (onMarkerPress) {
                 onMarkerPress(marker.id);
+
+                // find the index of the clicked marker in the locations array
+                const index = locations?.findIndex((loc) => loc.id === marker.id);
+
+                // if the index is found and is valid, scroll the FlatList to that index
+                if (index && index !== -1) {
+                  flatListRef.current?.scrollToIndex({ index, animated: true });
+                }
               }
             }}
           >
             <SvgXml xml={marker.icon} width={MARKER_ICON_SIZE} height={MARKER_ICON_SIZE} />
           </Marker>
         ))}
       </MapView>
+
       {isMaximizeButtonVisible && (
         <TouchableOpacity style={styles.maximizeMapButton} onPress={onMaximizeButtonPress}>
           <Icon.ExpandMap size={normalize(18)} />
         </TouchableOpacity>
       )}
+
+      {showPOIsFullScreenMap && (
+        <FlatList
```

Comment:
list logics were not requested at this time. but it is a cool way, we can get back to in the future. i saved it in a new branch https://github.com/smart-village-solutions/smart-village-app-app/tree/feature/MQGB-34-flat-list-markers-overlay